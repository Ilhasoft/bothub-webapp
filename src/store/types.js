export default {
  SET_TOKEN: 'SET_TOKEN',
  OPEN_LOGIN_MODAL: 'OPEN_LOGIN_MODAL',
  CLOSE_LOGIN_MODAL: 'CLOSE_LOGIN_MODAL',
  SET_PROFILE: 'SET_PROFILE',
  SET_NICKNAME_AUTHENTICATED: 'SET_NICKNAME_AUTHENTICATED',
  SET_REPOSITORY_RELATED_UUID: 'SET_REPOSITORY_RELATED_UUID',
  CACHE_FETCH: 'CACHE_FETCH',
};
