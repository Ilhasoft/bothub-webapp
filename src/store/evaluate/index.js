import actions from './actions';
import state from './state';
import mutations from './mutations';
import getters from './getters';
import types from './types';

export default {
  actions,
  state,
  mutations,
  getters,
  types,
};
