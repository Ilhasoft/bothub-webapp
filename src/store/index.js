import Vue from 'vue';
import Vuex from 'vuex';

import Auth from './auth';
import User from './user';
import Org from './org';
import Repository from './repository';
import Category from './category';
import Example from './example';
import EvaluateExample from './evaluate-example';
import Translate from './translate';
import CachedFetch from './cached-fetch';
import Entity from './entity';
import Intent from './intent';
import News from './news';
import Tutorial from './tutorial';
import External from './external';

Vue.use(Vuex);

const store = new Vuex.Store({
  modules: {
    Auth,
    User,
    Org,
    Repository,
    Category,
    Example,
    Translate,
    CachedFetch,
    EvaluateExample,
    Entity,
    News,
    Tutorial,
    External,
    Intent,
  },
});

store.dispatch('retriveAuthToken');

export default store;
