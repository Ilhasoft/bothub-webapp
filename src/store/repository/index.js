import state from './state';
import getters from './getters';
import actions from './actions';
import mutations from './mutations';
import types from './types';

export default {
  state,
  getters,
  actions,
  mutations,
  types,
};
