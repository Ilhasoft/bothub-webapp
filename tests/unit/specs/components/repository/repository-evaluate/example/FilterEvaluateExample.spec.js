import { shallowMount, createLocalVue } from '@vue/test-utils';
import FilterEvaluateExample from '@/components/repository/repository-evaluate/example/FilterEvaluateExample';
import store from '@/store';

import sinon from 'sinon';

const localVue = createLocalVue();


describe('TestView.vue', () => {
  let wrapper;
  let clock;
  beforeEach(() => {
    clock = sinon.useFakeTimers();
    wrapper = shallowMount(FilterEvaluateExample, {
      localVue,
      mocks: {
        $t: () => 'some specific text',
      },
      propsData: {
        intent: 'like',
        entity: 'cat',
        text: 'i like to eat sushi',
      },
      store,
    });
  });

  afterEach(() => {
    clock.restore();
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('emits querystringformatted event when intent data changes', () => {
    clock.tick(500);

    expect(wrapper.vm.$emit('querystringformatted')).toBeDefined();
  });

  test('emits querystringformatted event when entity changes', () => {
    clock.tick(500);
    expect(wrapper.vm.$emit('querystringformatted')).toBeDefined();
  });

  test('emits querystringformatted event when text changes', () => {
    clock.tick(500);
    expect(wrapper.vm.$emit('querystringformatted')).toBeDefined();
  });
});
