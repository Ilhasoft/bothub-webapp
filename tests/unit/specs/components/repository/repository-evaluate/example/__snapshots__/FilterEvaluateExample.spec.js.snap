// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestView.vue renders correctly 1`] = `
<div class="filter-evaluate-example" intent="like" entity="cat" text="i like to eat sushi">
  <div class="filter-evaluate-example__filters">
    <b-field-stub addons="true" class="filter-evaluate-example__filters__input-text">
      <b-input-stub usehtml5validation="true" statusicon="true" value="" type="text" hascounter="true" customclass="" iconright="magnify" debounce="750"></b-input-stub>
    </b-field-stub>
    <div class="filter-evaluate-example__filters__wrapper filter-evaluate-example__filters__wrapper__has-1-fields">
      <div class="filter-evaluate-example__filters__wrapper__text">
        some specific text:
      </div>
      <b-field-stub addons="true">
        <b-autocomplete-stub usehtml5validation="true" statusicon="true" value="" data="" field="value" openonfocus="true" dropdownposition="bottom" confirmkeys="Tab,Enter" formatters="inputFormatters" placeholder="some specific text"></b-autocomplete-stub>
      </b-field-stub>
      <!---->
      <!---->
      <b-field-stub addons="true">
        <!---->
      </b-field-stub>
    </div>
  </div>
</div>
`;
